assembl/views/api/post.py

```diff
@@ -313,16 +373,68 @@ def get_posts(request):
         posts = posts.order_by(Content.body_text_index.score_name.desc())
     else:
         posts = posts.order_by(Content.id)
-    print str(posts)
+    # print str(posts)
 
     no_of_posts = 0
     no_of_posts_viewed_by_user = 0
 
+
+    def post_and_its_whole_line_of_descent_are_deleted(post):
+        """
+        :returns: True if post and its whole line of descent (direct and indirect answers) are deleted, False otherwise.
+        :rtype: bool
+        :param post: The post you want to analyse
+        :type post: assembl.models.Post
+
+        """
+        if not post.is_tombstone:
+            return False
+        children = post.children
+        if len(children) == 0:
+            return True
+        for child in children:
+            if not post_and_its_whole_line_of_descent_are_deleted(child):
+                return False
+        return True
+
+    def post_or_one_of_line_of_descent_is_deleted(post):
+        """
+        :returns: True if post or one of its line of descent (direct and indirect answers) is deleted, False otherwise.
+        :rtype: bool
+        :param post: The post you want to analyse
+        :type post: assembl.models.Post
+        
+        """
+        if post.is_tombstone:
+            return True
+        children = post.children
+        if len(children) == 0:
+            return False
+        for child in children:
+            if post_or_one_of_line_of_descent_is_deleted(child):
+                return True
+        return False
+
     for query_result in posts:
         score, viewpost, likedpost = None, None, None
         if not isinstance(query_result, (list, tuple)):
             query_result = [query_result]
         post = query_result[0]
+
+        # The response should not include deleted posts which do not break the structure of threads (these are deleted posts which have not received any direct or indirect non-deleted answer)
+        # Look for non-deleted (direct or indirect) children, and if there is none, we know that removing it from the results will not break the structure
+        # TODO: use recursion at the SQL query level instead
+        ignore_this_post = False
+        if deleted == 'false':
```

Comment:
Not sure to understand why we use Strings instead of boolean. Is there a reason I can't see yet? In case we would use booleans, the second test would be useless and we could just replace it by "else".